### DISCUSSION TITLE
Need to override getNextSocketAddress for failover due to "Disconnecting : timed out " from SocketConnectPort to SocketConnectPort1

### DISCUSSION THREAD

**Discussion by B-a-l-a-j-i-s** [quickfix-j/quickfixj#622] [General]
I want to do the custom failover for the socket timeout issue by overriding the IoSessionInitiator , getNextSocketAddress. Please give the code to implement or any other way to achieve it.

**Comment by chrjohn** [comment]
I'm not sure I understand what you mean. Probably it is related to this: https://github.com/quickfix-j/quickfixj/issues/402 ??

**Reply by B-a-l-a-j-i-s** [reply]
Hi @chrjohn , it is the similar issue, It throws error : Disconnecting: Timed out waiting for logon response, but it's not failover from SocketConnectPort to SocketConnectPort1. Is there any change to be done programmatically to achieve. Please share your thoughts.

**Reply by chrjohn** [reply]
Did you read the linked issue and the mailing list thread linked in that issue? It suggests to create a interface or strategy which then could be used to implement a different failover behaviour.
